app.py

```diff
@@ -0,0 +1,136 @@
+TOKEN_URL = os.getenv("RSO_BASE_URI") + "/token"
+LOGOUT_URL = os.getenv("RSO_BASE_URI") + "/logout"
+APP_CALLBACK_URL = os.getenv("APP_BASE_URL") +":"+ os.getenv("PORT") + os.getenv("APP_CALLBACK_PATH")
+RSO_CLIENT_ID = os.getenv("RSO_CLIENT_ID")
+AUTH_URL=AUTHORIZE_URL + "?redirect_uri=" + APP_CALLBACK_URL + "&client_id=" + RSO_CLIENT_ID + "&response_type=code" + "&scope=openid"
+
+@app.route("/", methods=["POST","GET"])
+def index(): 
+    return render_template("index.html", AUTH_URL=AUTH_URL, LOGOUT_URL=LOGOUT_URL)
+
+@app.route("/oauth-callback", methods=["POST","GET"])
+def aouthCallback():
+    code = request.args.get("code")
+
+    data = {
+    "grant_type": "authorization_code",
+    "code": code,
+    "redirect_uri": APP_CALLBACK_URL
+    }
+
+    try:
+        req = requests.Session()
+        req.auth = (os.getenv("RSO_CLIENT_ID"), os.getenv("RSO_CLIENT_SECRET"))
+
+        r = req.post(
+            url = TOKEN_URL, 
+            data = data, 
+            headers={"Content-Type":"application/x-www-form-urlencoded"},
+            )
+
+        session["token"] = r.json().get("access_token")
+
+        return render_template("main.html", account={})   
+    except Exception as e:
+        print(f"Error token not generated {e}")
+
+        return render_template("index.html", AUTH_URL=AUTH_URL, LOGOUT_URL=LOGOUT_URL)   
+
+@app.route("/account", methods=["POST","GET"])
+def account():
+
+    if not session.get("token"): 
+        return jsonify("Unauthorized"), 401   
+
+    if request.method == "POST":
+        requestURL = f"https://{os.getenv('REGION')}.api.riotgames.com/riot/account/v1/accounts/me"
+        header = {"Authorization": "Bearer " + session["token"]}
+
+        try:
+            response = requests.get(requestURL, headers=header)
+
+            if response.status_code != 200: 
+                return render_template("main.html", ACCOUNT_INFO=response.json().get("status")["message"])
+
+            
+            return render_template("main.html", ACCOUNT_INFO=response.json())
+        except:
+
+            return render_template("main.html", ACCOUNT_INFO="Error getting account info ")
+
+@app.route("/accounts/me", methods=["GET"])
+def accountMe():
+
+    if not session.get("token"): 
+        return jsonify("Unauthorized"), 401   
+
+    requestURL = f"https://{os.getenv('REGION')}.api.riotgames.com/riot/account/v1/accounts/me"
+    header = {"Authorization": "Bearer " + session.get("token")}
+
+    try:
+        response = requests.get(requestURL, headers=header)
+
+        if response.status_code != 200: 
+            return jsonify("Internal Server Error"), 500  
+
+        return response.json(), 200  
+    except Exception as e:
+            print(e)
+            return jsonify("Internal Server Error"), 500  
+
+@app.route("/summoners/overview", methods=["GET"])
+def summonersOverview():
+
+    region  = request.args.get("region", None)
+    name  = request.args.get("name", None)
```

Comment:
I might use the variable `summonerName` here and in line 107 just to be more specific in the code, since we can't change the api